### ISSUE TITLE
Disable looking in parent context for value?

### ISSUE THREAD

**Issue by sethladd** [valotas/mustache4dart#41]
Hi,

Consider this case:

```
class Parent {
  String foo;
}

class Child extends Parent {
  List<OtherChild> children = [];
}

class OtherChild extends Parent {

}  

main() {
var c = new Child()
        ..foo = 'child'
        ..children = [new OtherChild()..foo='otherchild', new OtherChild()..foo=null];
}
```

(notice the second instance of OtherChild has a null foo.

Here is the template:

```
{{foo}}
{{#children}}{{foo}}!{{/children}}
```

```
Expected: 'child\n'
    'otherchild!!'
    Actual: 'child\n'
    'otherchild!child!'
     Which: is different.
  Expected: ... therchild!!
    Actual: ... therchild!child!
```

MustacheContext does this:

```
  operator [](String key) {
    if (ctx == null) return null;
    return _getInThisOrParent(key);
  }
```

As you can see, it tries to get a name from the current context, or the parent context.

I believe the correct behavior should be: "if a context has a field for a value, but that value is null, stop searching and return null. Don't treat null the same as _the field doesn't exist_"

I'll work on a patch to see what's up.

**Comment by sethladd** [comment]
Here is a test case:

```
    test('#XX do not look into parent context if current context has field but its value is null', () {
      var p = new Parent()
        ..foo = 'parent'
        ..children = [new Child()..foo='child', new Child()..foo=null];

      var template = '''
{{foo}}
{{#children}}{{foo}}!{{/children}}''';

      var output = render(template, p);
      var expected = "parent\nchild!!";

      expect(output, expected);
    });
```

**Comment by sethladd** [comment]
Fix and tests are here: https://github.com/sethladd/mustache4dart/commit/c3f60253e90192365f8d97a9b3a9be9df12ce647

All tests pass.

**Comment by valotas** [comment]
The truth is that it is questionable how you'd like to treat nulls. But I see a valid case here.

I'll merge this as soon as I'll get home.

Thanks

**Comment by sethladd** [comment]
Much appreciated

**Comment by valotas** [comment]
Within 562dd5cb631496ce78950b09ce1be73685aa2279 I've introduced `assumeNullNonExistingProperty` which is by default `true` to preserve the previous behavior. Just call `render` with `assumeNullNonExistingProperty: false` and your are done.

**Comment by valotas** [comment]
If you need the changes, I can push a new release today. Otherwise I'll try to fix #39 first

**Comment by sethladd** [comment]
Thanks! I'm not blocked because I'm using my fork, so feel free to fix #39 first. Then, I can get rid of my fork :)

Really appreciate the responsive help.
